execution/scan/matrix_selector.go

```
@@ -448,5 +455,12 @@ loop:
 		}
 	}
 
+	if selectsNativeHistograms {
+		warnings.AddToContext(
```

Maybe it's better to simply return an error here, so that the edge case is handled in one place.